vespaclient-container-plugin/src/main/java/com/yahoo/vespa/http/server/FeedReaderFactory.java

```diff
@@ -31,7 +33,14 @@ public FeedReader createReader(
                 try {
                     return new VespaXMLFeedReader(inputStream, docTypeManager);
                 } catch (Exception e) {
-                    throw new RuntimeException("Could not create VespaXMLFeedReader", e);
+                    byte [] peek;
+                    try {
+                        peek = new byte[Math.min(200, inputStream.available())];
+                        inputStream.read(peek);
+                    } catch (IOException  io) {
+                        peek = new byte [0];
+                    }
+                    throw new RuntimeException("Could not create VespaXMLFeedReader. First characters are: " + Utf8.toString(peek), e);
```

Comment:
Will the bytes contain the actual first bytes? I would assume that the `VespaXMLFeedReader` consumes some bytes before throwing the exception.

Reply:
You are right. I added mark/reset support and made it default to peek the first 200 bytes.
Will consider if this should become a feature flag.